src/Tests/dotnet-workload-install.Tests/GivenWorkloadManifestUpdater.cs

```diff
@@ -320,14 +321,27 @@ public void GivenWorkloadsAreOutOfDateUpdatesAreAdvertisedOnRestoringCommands(st
             File.WriteAllText(Path.Combine(testInstance.Path, ".dotnet", ".workloadAdvertisingManifestSentinal"), string.Empty);
 
             var command = new DotnetCommand(Log);
-            command
+            var commandResult = command
                 .WithEnvironmentVariable("DOTNET_CLI_HOME", testInstance.Path)
                 .WithWorkingDirectory(testInstance.Path)
-                .Execute(commandName)
-                .Should()
-                .Pass()
-                .And
-                .HaveStdOutContaining(Workloads.Workload.Install.LocalizableStrings.WorkloadUpdatesAvailable);
+                .Execute(commandName);
+            if (!commandOptedOut)
+            {
+                commandResult
+                    .Should()
+                    .Pass()
+                    .And
+                    .HaveStdOutContaining(Workloads.Workload.Install.LocalizableStrings.WorkloadUpdatesAvailable);
+            }
+            else
+            {
+                commandResult
+                    .Should()
+                    .Pass()
+                    .And
+                    .NotHaveStdOutContaining(Workloads.Workload.Install.LocalizableStrings.WorkloadUpdatesAvailable);
+            }
```

Comment:
I think you can do the following if you want to deduplicate the code that checks that the command passed.

```suggestion
            commandResult
                .Should()
                .Pass()

            if (!commandOptedOut)
            {
                commandResult
                    .Should()
                    .HaveStdOutContaining(Workloads.Workload.Install.LocalizableStrings.WorkloadUpdatesAvailable);
            }
            else
            {
                commandResult
                    .Should()
                    .NotHaveStdOutContaining(Workloads.Workload.Install.LocalizableStrings.WorkloadUpdatesAvailable);
            }
```